pkg/archive/archive.go

```diff
@@ -205,6 +222,11 @@ func ExtractArchiveToTempDir(ctx context.Context, path string) (string, error) {
 		return true
 	})
 
+	if ctx.Err() != nil {
+		defer os.RemoveAll(tmpDir)
+		return "", ctx.Err()
+	}
```

Comment:
I think you can remove this - the RemoveAll is redundant with the goroutine above, and at this point we've already done all of the expensive extraction so you might as well return normally (another caller that's sensitive to context cancellations will likely do a similar check)

Reply:
Removed in [`62a0438` (#876)](https://github.com/chainguard-dev/malcontent/pull/876/commits/62a043878f3ddc344f08d705c83b08e1c9425688).